### PR TITLE
[mtg-576] Ultra experimental penalties

### PR DESCRIPTION
**Pull request by kstepanovdev**

### CODE REVIEW

File: program-states/src/state/voter.rs [hunk 1]
```diff
@@ -68,6 +72,34 @@ impl Voter {
...
+            Err(MplStakingError::MiningAlreadyRestricted.into())
+        } else {
+            self.penalties |= Self::IS_TOKENFLOW_RESTRICTED_MASK;
+            Ok(())
+        }
+    }
+
+    pub fn allow_tokenflow(&mut self) -> Result<()> {
+        if !self.is_tokenflow_restricted() {
+            Err(MplStakingError::MiningAlreadyRestricted.into())
```
Comment (StanChe) [inline program-states/src/state/voter.rs:87]: Probably a different error (or rename this one) as this one is misleading

File: programs/mpl-staking/src/instructions/penalties/restrict_batch_minting.rs [hunk 2]
```diff
@@ -17,17 +13,13 @@ pub fn restrict_batch_minting(
         MplStakingError::InvalidRealmAuthority
     );
 
-    let signers_seeds = registrar_seeds!(&registrar);
+    require!(
+        until_ts > clock_unix_timestamp(),
```
Comment (StanChe) [inline programs/mpl-staking/src/instructions/penalties/restrict_batch_minting.rs:17]: I think we could drop this check as a way to remove the restriction. The use case is to restrict a user for a month and then remove the restriction once the user is cleared

**Review by StanChe — APPROVED** [review]
Great job! Just couple minor fixes

File: programs/mpl-staking/src/instructions/penalties/restrict_batch_minting.rs (hunk 2 above)
Comment (kstepanovdev) [reply programs/mpl-staking/src/instructions/penalties/restrict_batch_minting.rs:17]: It's a typo. It rather should've been `until_ts < clock_unix_timestamp()` in order not to set the time that is lower than the current one.

**Pull request merged**